### ISSUE TITLE
"Invalid data" error when writing result.stderr to process.stderr

### ISSUE THREAD

**Issue by idpaterson** [ForbesLindesay/spawn-sync#8]
Commit 5d1d0b9 introduces a bug in cases where the spawned process does not write to stderr or stdout. Rather than an empty buffer, `result.stdout` and `result.stderr` are represented by an empty array. Attempting to write the array to an output stream fails with:
```
TypeError: invalid data
    at WriteStream.Socket.write (net.js:612:11)
```

### Test case

Please add a test case that writes to only one or neither of stderr and stdout. The result can be verified with an assertion like `Buffer.isBuffer(result.stderr)`.

### Temporary workaround

I have temporarily avoided this by fencing any write operations with a check against the buffer (or in this case, array) length:
```js
if (result.stderr.length > 0) {
    process.stderr.write(result.stderr);
}
```

**Comment by ForbesLindesay** [comment]
Thanks.  Sorry it took a while to get round to looking at this.  It should be fixed in 1.0.3

**Comment by idpaterson** [comment]
Thank you! Now I can take down my `.length > 0` tests.
